types/Account.d.ts

```diff
@@ -0,0 +1,76 @@
+export declare class Account {
+    /** @property {string} username */
+    username: string;
+    /** @property {Array<string>} characters List of character names in this account */
+    characters: Array<string>;
+    /** @property {string} password Hashed password */
+    password: string;
+    /** @property {boolean} banned Whether this account is banned or not */
+    banned: boolean;
+
+    /**
+     * @param {Object} data Account save data
+     */
+    constructor(data: Object);
```

Comment:
The Object here could potentially be declared as an interface with the following keys:

```
    this.username   = data.username;
    this.characters = data.characters || [];
    this.password   = data.password;
    this.banned = data.banned || false;
    this.deleted = data.deleted || false;
    // Arbitrary data bundles are free to shove whatever they want in
    // WARNING: values must be JSON.stringify-able
    this.metadata = data.metadata || {};
```

Since most of these are also defined as properties, I wonder if TypeScript has a way of making this DRY, e.g. define an interface, define the constructor as accepting an object with that interface, and also note that the class also implements that same interface. Thoughts?

Reply:
I think is a good idea like `ChannelOptions` or `ChannelAudience` that has been defined only on TS side as interfaces:

Reply:
We can define an interface with that properties only and the required type for Account constructor should be that type instead a generic Object